backtracking/rat_in_maze.py

```diff
@@ -1,81 +1,125 @@
 from __future__ import annotations
 
 
-def solve_maze(maze: list[list[int]]) -> bool:
+def solve_maze(
+    maze: list[list[int]],
+    source_row: int,
+    source_column: int,
+    destination_row: int,
+    destination_column: int,
+) -> list[list[int]] | None:
```

Comment:
Returning two different datatypes pushes work on the caller.  Please consider raising an exception if there is no solution.

Reply:
See my remark in the general discussion of this Pull Request.

Reply:
```suggestion
) -> list[list[int]]:
```